### ISSUE TITLE
Enable parallel feature by default

### ISSUE THREAD

**Issue by aminya** [rust-lang/cc-rs#700]
It is not practical to go through every single dependency and enable the parallel feature for them. This should be a default feature.

**Comment by nathaniel-daniel** [comment]
Wouldn't enabling this feature once for this library enable it for everything though? It seems worse to force people to go to each dependency and find the feature flag to disable cc's default features, assuming they even expose such functionality.

**Comment by nathaniel-daniel** [comment]
True, but enabling the parallel feature will also pull in a new dependency. If this dependency is unwanted, it becomes much harder to remove it since default features are difficult to disable.

**Comment by aminya** [comment]
So, you mean people want to make the build n times slower just not to grab such a small build dependency?

**Comment by aminya** [comment]
It is not possible to enable a feature when you run `cargo install` . Run `cargo install outdated` and wait for the build to happen very slowly. 

The time it takes to build job server is nothing when compared to the time it takes to build `cargo`. When designing a tool, you should consider the average experience and not the exceptions. In most cases, the parallel build has visible benefits.

**Comment by nathaniel-daniel** [comment]
It is possible to enable features for a `cargo install` build with the `--features` flag. One could add a feature to their binary to easily allow users to make their build parallel while installing, even by default. 

I'm assuming you mean `cargo install cargo-outdated` when you say `cargo install outdated`. That build should already have parallel enabled; libgit2-sys turns it on since it has a large number of files that need to be built. I think this is how it should be: libraries that have a large number of files to compile with cc can opt-in to a faster, parallel build (forcing it for all other crates as well). 

I'm also not sure that a parallel build is a good for for the average experience. Some crates rely on cc just to find build tools, like cmake or jemalloc-sys. Others only need to compile a few files, like signal-hook and libsqlite3-sys. Others are forced to use it as a form of platform detection, like backtrace or findshlibs. I'm just not sure making parallel the default is a good idea as cc is used as more than a method of compiling large numbers of c files. For a ubiquitous dependency like cc, enabling the parallel feature by default would almost be the same as removing the feature and merging the functionality into cc.
